src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/BuildTimeController.cs

```diff
@@ -0,0 +1,128 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Kusto.Data.Common;
+using Maestro.Data;
+using Maestro.Web.Api.v2019_01_16.Models;
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.DotNet.Kusto;
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Data;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+
+namespace Maestro.Web.Api.v2019_01_16.Controllers
+{
+    /// <summary>
+    ///   Exposes methods to Read <see cref="BuildTime"/>s
+    /// </summary>
+    [Route("buildtime")]
+    [ApiVersion("2019-01-16")]
+    public class BuildTimeController : ControllerBase
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        private readonly IKustoClientProvider _kustoClientProvider;
+
+        public BuildTimeController(
+            BuildAssetRegistryContext context,
+            IKustoClientProvider kustoClientProvider)
+        {
+            _context = context;
+            _kustoClientProvider = kustoClientProvider;
+        }
+
+        /// <summary>
+        /// Gets the average official build time and average pr build time for a given default channel
+        /// This is captured for generating the longest build times for the dependency flow graph
+        /// </summary>
+        /// <param name="id">Default channel id</param>
+        [HttpGet("{id}")]
+        [SwaggerApiResponse(HttpStatusCode.OK, Type = typeof(BuildTime), Description = "Gets the average official build time and average pr build time for a given default channel")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> GetBuildTimes([Required]int id)
+        {
+            Data.Models.DefaultChannel defaultChannel = await _context.DefaultChannels.FindAsync(id);
+
+            if (defaultChannel == null)
+            {
+                return NotFound();
+            }
+
+            var properties = new ClientRequestProperties();
+
+            // There is a mismatch between how we store the names of the repositories in BAR
+            // and how we store them in Kusto. Normalize to the last part of the url.
+            properties.SetParameter("_Repository", defaultChannel.Repository.Split("/").Last());
```

Comment:
I considered porting over the KustoQuery class from Helix to Maestro, but didn't. I think it would clean up the code, but wasn't sure if that would be overkill. Thoughts?

Reply:
I think we've got a fair bit of it already, it couldn't hurt, and it can abstract away a lot of the Kusto weirdness that can be tricky to get right, since you've go to always do 3 things to make it work (and types are a thing)